src/handlers/courtesy.ts

```diff
@@ -28,21 +20,37 @@ import { dbInstance } from "../utils/cosmos";
 import { log } from "../utils/logger";
 import { Client } from "../generated/client";
 import { ActivationStatusEnum } from "../generated/ActivationStatus";
-import { validateTaxIdHeader } from "../utils/validators";
+import { validateTaxIdInHeader } from "../utils/validators";
+import {
+  IPNResponseErrorForbiddenNotAuthorized,
+  IPNResponseErrorInternal,
+  PNResponseErrorForbiddenNotAuthorized,
+  PNResponseErrorInternal,
+  PNResponseErrorNotFound,
+  PNResponseErrorValidation,
+  IPNResponseErrorValidation,
+  IPNResponseErrorNotFound
+} from "../utils/responses";
 
 export const courtesyGetHandler = (
   req: express.Request
 ): Promise<
-  | IResponseErrorValidation
-  | IResponseErrorNotFound
-  | IResponseErrorForbiddenNotAuthorized
-  | IResponseErrorInternal
+  | IPNResponseErrorValidation
+  | IPNResponseErrorNotFound
+  | IPNResponseErrorForbiddenNotAuthorized
+  | IPNResponseErrorInternal
   | IResponseSuccessJson<IoCourtesyDigitalAddressActivation>
 > =>
   pipe(
     req.headers["x-api-key"],
-    TE.fromNullable(ResponseErrorForbiddenNotAuthorized),
-    TE.chainW(validateTaxIdHeader(req)),
+    TE.fromNullable(
+      PNResponseErrorForbiddenNotAuthorized(
+        "Forbidden not authorized",
+        "The x-api-key was not provided",
+        []
+      )
+    ),
+    TE.chainW(() => validateTaxIdInHeader(req)),
```

Comment:
`validateTaxIdInHeader` should correctly return a left if `req.headers["x-api-key"]` is nullable, so I would remove `TE.fromNullable`

Reply:
nvm i got confused with the `validateTaxIdInHeader`, however i would move the `TE.fromNullable` to a function like `validateXApiKey`